internal/worker/server.go

```diff
@@ -309,6 +309,12 @@ func (s *Server) OSBuildJobInfo(id uuid.UUID, result *OSBuildJobResult) (*JobInf
 			result.JobError = clienterrors.WorkerClientError(clienterrors.ErrorTargetError, "at least one target failed", result.TargetErrors())
 		}
 	}
+	if result.JobError != nil && result.JobError.ID == clienterrors.ErrorBuildJob {
+		rpmError := result.RpmError()
+		if rpmError != nil {
+			result.JobError = rpmError
+		}
+	}
```

Comment:
Can we just call the RpmError before setting the `result.JobError` and then set it in the details directly?

above:
```
result.JobError = clienterrors.WorkerClientError(clienterrors.ErrorBuildJob, "osbuild build failed", nil)
```

Reply:
In that particular case, the `JobError` is already set in the DB entry, so we have to check for it I think.

Reply:
I mean setting the string directly in details in line 305, right above this block.

Reply:
I'm sorry, I still don't understand. Line `305` is in a case where the `JobError` from the DB is `nil`. Where in the opposite case. What am I missing ?